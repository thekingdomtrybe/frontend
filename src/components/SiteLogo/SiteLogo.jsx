import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import SVG from '@/components/SVG/SVG';
import Styles from './SiteLogo.module.scss';

function SiteLogo({
  variant,
  live,
  fontClass,
}) {
  let className = Styles.logo;
  className += ` ${fontClass}`;

  switch (variant) {
    case 'blue': {
      className = `${className} ${Styles.blue}`;
      break;
    }
    case 'dark': {
      className = `${className} ${Styles.dark}`;
      break;
    }
    default:
      break;
  }

  if (live) className = `${className} ${Styles.live}`;
  const logoColor = variant === 'light' ? 'var(--dark)' : 'var(--white)';

  return (
    <Link to="/" className={className}>
      <SVG icon="favicon" color={logoColor} />
      <span>
        {
          live ? 'LIVE' : 'THE KINGDOM TRYBE'
        }
      </span>
    </Link>
  );
}

SiteLogo.propTypes = {
  variant: PropTypes.string,
  live: PropTypes.bool,
  fontClass: PropTypes.string,
};

SiteLogo.defaultProps = {
  variant: 'light',
  live: false,
  fontClass: '',
};

export default SiteLogo;
