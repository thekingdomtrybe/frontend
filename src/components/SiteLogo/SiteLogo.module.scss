.logo {
  font-family: "Expressa Regular", sans-serif;
  font-size: var(--size-4);
  text-decoration: none;
  color: var(--black);

  display: grid;
  grid-gap: var(--spacing-1);
  grid-template-columns: var(--size-7) auto;
  align-items: center;
  justify-content: center;
}

.blue,
.dark {
  color: var(--white);
}

.live {
  font-family: 'Roboto', Arial, sans-serif;
  font-weight: 800;
  color: var(--dark);
}
