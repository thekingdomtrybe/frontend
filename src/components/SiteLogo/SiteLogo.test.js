import renderer from 'react-test-renderer';
import SiteLogo from './SiteLogo.jsx';

it('renders correctly', () => {
  const component = renderer.create(<SiteLogo />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
