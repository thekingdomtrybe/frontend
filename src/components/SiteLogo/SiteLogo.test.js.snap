// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="logo"
>
  <img
    alt=""
  />
  <span>
    THE KINGDOM TRYBE
  </span>
</div>
`;
